### PR TITLE
Added routing module, metric selection is now routable

### PR DESCRIPTION
**Pull request by mateuszmidor**
Routing related code moved to routing.py
Metric selection now updates the url

### CODE REVIEW

**Review by Fenthick — DISMISSED** [review]

File: presentation/chart_view.py [hunk 1]
```diff
@@ -31,7 +27,7 @@ def make_layout(self, from_agent, to_agent: AgentID, mesh: MeshResults) -> html.
         content.append(
             html.Div(
                 html.Center(
-                    dcc.Link("Back to matrix view", href="/"),
+                    dcc.Link("Back to matrix view", href=back_to),
```
Comment (mmac-m3a) [inline presentation/chart_view.py:30]: Just remove it :)

File: routing.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def encode_matrix_path(metric: MetricType) -> str:
+    if metric == MetricType.PACKET_LOSS:
+        return "/packet_loss"
+    if metric == MetricType.JITTER:
+        return "/jitter"
+    if metric == MetricType.LATENCY:
+        return "/latency"
+    return "/"
```
Comment (mmac-m3a) [inline routing.py:15]: We need to log an error here, because this should never happen.

File: routing.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def encode_chart_path(from_agent, to_agent: AgentID, return_path: str) -> str:
+    return f"/chart?from={from_agent}&to={to_agent}&backto={return_path}"
```
Comment (mmac-m3a) [inline routing.py:19]: Will the "backto" be visible in the history?

**Review by mmac-m3a — APPROVED** [review]
Routing changes make the code better. It is though IMHO still unnecessarily complicated. If we normalize the path to contain the type of view and query parameters to contain what to render, routing can consist of a route table (dictionary mapping path to a layout rendering function) and a single function. We really have only 2 paths:
```
/matrix?metric=<metric_name>
/chart?from=<agent_id>&to=<agent_id>
```
and `/` which is just alias for `/matrix?metric=latency`.

Better frameworks provide a decorator that builds the routing table and do the routing automatically ;-).

File: main.py [hunk 4]
```diff
@@ -93,29 +99,27 @@ def run_development_server(self) -> None:
...
 
-    def _make_matrix_layout(self) -> html.Div:
+    def _make_matrix_layout(self, metric: MetricType) -> html.Div:
         mesh_test_results = self._cached_repo.get_mesh_test_results()
-        metric = self._current_metric
         return self._matrix_view.make_layout(mesh_test_results, metric, self._config)
 
     def _make_chart_layout(self, path: str) -> html.Div:
-        from_agent, to_agent = ChartView.decode_path(path)
+        from_agent, to_agent, back_to = routing.decode_chart_path(path)
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply main.py:120]: Yes we can

File: routing.py (hunk 3 above)
Comment (mateuszmidor) [reply routing.py:19]: Removing back button together with backto, so - no :)

File: main.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply main.py]: Updated.

File: presentation/chart_view.py (hunk 1 above)
Comment (mateuszmidor) [reply presentation/chart_view.py]: Done.

File: routing.py (hunk 2 above)
Comment (mateuszmidor) [reply routing.py]: Done.

**Review by Fenthick — APPROVED** [review]

**Review by mmac-m3a — APPROVED** [review]
I like this version!

**Pull request merged**